lib/ansible/cli/adhoc.py

```diff
@@ -64,10 +64,18 @@ def post_process_args(self, options):
 
         return options
 
-    def _play_ds(self, pattern, async_val, poll):
-        check_raw = context.CLIARGS['module_name'] in C.MODULE_REQUIRE_ARGS
-
-        mytask = {'action': {'module': context.CLIARGS['module_name'], 'args': parse_kv(context.CLIARGS['module_args'], check_raw=check_raw)},
+    def _play_ds(self, pattern, async_val, poll, loader):
+        if context.CLIARGS['module_args'].startswith(u"@"):
```

Comment:
at this point, just use a playbook

Reply:
I agree with removal, I added it so it matches the extra vars, but I'm a bit afraid there will be some more edge cases